frameworks/cassandra/src/main/java/com/mesosphere/sdk/cassandra/scheduler/ZoneValidator.java

```diff
@@ -0,0 +1,80 @@
+package com.mesosphere.sdk.cassandra.scheduler;
+
+import com.mesosphere.sdk.config.validate.ConfigValidationError;
+import com.mesosphere.sdk.config.validate.ConfigValidator;
+import com.mesosphere.sdk.offer.TaskUtils;
+import com.mesosphere.sdk.specification.ServiceSpec;
+import com.mesosphere.sdk.specification.TaskSpec;
+import org.apache.logging.log4j.util.Strings;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Optional;
+
+/**
+ * This class validates that the DETECT_ZONES envvar can only suppor the following transitions.
+ *
+ * 1. null  --> false
+ * 2. true  --> true
+ * 3. false --> false
+ */
+public class ZoneValidator implements ConfigValidator<ServiceSpec> {
+    static final String POD_TYPE = "node";
+    static final String TASK_NAME = "server";
+    static final String DETECT_ZONES_ENV = "DETECT_ZONES";
+
+    @Override
+    public Collection<ConfigValidationError> validate(Optional<ServiceSpec> oldConfig, ServiceSpec newConfig) {
+        if (!oldConfig.isPresent()) {
+            return Collections.emptyList();
+        }
+
+        Optional<TaskSpec> oldTask = TaskUtils.getTaskSpec(oldConfig.get(), POD_TYPE, TASK_NAME);
+        if (!oldTask.isPresent()) {
+            // Maybe the pod or task was renamed? Lets avoid enforcing whether those are rename- able and assume it's OK
```

Comment:
These situations (where something has an identity between the previous and current spec and we maybe lose track of it) kind of keep me up at night. Just a note for posterity that it'd be nice to figure out how to surface those changes explicitly.